src/mono/mono.proj

```diff
@@ -919,19 +920,30 @@ JS_ENGINES = [NODE_JS]
     </PropertyGroup>
 
     <PropertyGroup>
-      <MonoLibClang Condition="$([MSBuild]::IsOSPlatform('OSX')) and '$(MonoLibClang)' == ''">$(XcodeDir)/Toolchains/XcodeDefault.xctoolchain/usr/lib/libclang.dylib</MonoLibClang>
-      <MonoLibClang Condition="$([MSBuild]::IsOSPlatform('Linux')) and '$(MonoLibClang)' == ''">@(_LibClang)</MonoLibClang>
-      <MonoLibClang Condition="'$(HostOS)' == 'windows' and '$(MonoLibClang)' == ''">c:/dev/LLVM/bin/libclang.dll</MonoLibClang>
       <MonoAotCMakeSysroot Condition="'$(MonoAotCMakeSysroot)' == ''">$(MonoCrossDir)</MonoAotCMakeSysroot>
     </PropertyGroup>
+
+    <ItemGroup Condition="'$(MonoUseCrossTool)' == 'true'">
+      <_LibclangPackageReferencePathPropertyName Include="Pkgruntime_$(_portableHostOS)-$(BuildArchitecture)_Microsoft_NETCore_Runtime_Mono_LLVM_Libclang" />
+    </ItemGroup>
+    <PropertyGroup Condition="'$(MonoUseCrossTool)' == 'true'">
+      <_LibclangPackageReferencePath>$(%(_LibclangPackageReferencePathPropertyName.Identity))</_LibclangPackageReferencePath>
+      <_LibclangSuffix Condition="'$(HostOS)' == 'Windows'">.dll</_LibclangSuffix>
+      <_LibclangSuffix Condition="'$(HostOS)' == 'OSX'">.dylib</_LibclangSuffix>
+      <_LibclangSuffix Condition="'$(_LibclangSuffix)' == ''">.so</_LibclangSuffix>
+      <_LibclangPath>$(_LibclangPackageReferencePath)\tools\libclang$(_LibclangSuffix)</_LibclangPath>
```

Comment:
```suggestion
      <_LibclangPath>$(_LibclangPackageReferencePath)\tools\libclang$(LibSuffix)</_LibclangPath>
```
(eng/native/naming.props is already included in all contexts)

Reply:
that doesn't work since we need the extension for the host OS where the offsets-generator runs. naming.props is based off of PackageRID i.e. TargetOS

Reply:
ah, right, maybe we can create another set there completely based on host if this kind of thing has more than one uses (e.g. CG2 may also be doing similar shenanigans somewhere 😅).